alacritty/src/event.rs

```diff
@@ -1310,6 +1337,27 @@ impl Processor {
             }
 
             match event {
+                // The event loop just got initialized. Create a window.
+                GlutinEvent::Resumed => {
+                    // Creating window inside event loop is required for platforms like macOS to
+                    // properly initialize state, like tab management. Othwerwise the first window
+                    // won't handle tabs.
+                    let initial_window_options = match initial_window_options.take() {
+                        Some(initial_window_options) => initial_window_options,
+                        None => return,
+                    };
+
+                    if let Err(result) =
+                        self.create_window(event_loop, proxy.clone(), initial_window_options)
+                    {
+                        // Stash the error and exit with non-zero code.
+                        *loop_error.lock().unwrap() = result.into();
+                        *control_flow = ControlFlow::ExitWithCode(1);
+                        return;
```

Comment:
Wait can we not just exit with any generic result, which is then returned from the closure?! That's kinda garbage.

Let's maybe just print the error directly inside the `run_return`, then exit upwards with just any random error text?

I **definitely** don't want to throw mutexes at a silly one-off error that can only happen right before the event loop crashes anyway.

Reply:
We can sure.